libr/lang/p/spp_r2.h

```diff
@@ -0,0 +1,470 @@
+		if (p) {
+			b = p;
+			strcat (b, "/");
+			strcat (b, buf);
+			spp_file (b, out);
+		}
+		free (b);
+	} else {
+		spp_file(buf, out);
+	}
+	return 0;
+}
+
+static TAG_CALLBACK(spp_r2_if) {
+	char *var = spp_r2_var_get(buf);
+	state->echo[state->ifl + 1] = (var && *var != '0' && *var != '\0') ? 1 : 0;
+	return 1;
+}
+
+/* {{ ifeq $path / }} */
+static TAG_CALLBACK(spp_r2_ifeq) {
+	char *value = buf;
+	char *eq = strchr(buf, ' ');
+	if (eq) {
+		*eq = '\0';
+		value = spp_r2_var_get(value);
+		if (value && !strcmp(value, eq+1)) {
+			state->echo[state->ifl + 1] = 1;
+		} else state->echo[state->ifl + 1] = 0;
+//fprintf(stderr, "IFEQ(%s)(%s)=%d\n", buf, eq+1, echo[ifl]);
+	} else {
+		value = spp_r2_var_get(buf);
+		if (!value || *value=='\0')
+			state->echo[state->ifl + 1] = 1;
+		else state->echo[state->ifl + 1] = 0;
+//fprintf(stderr, "IFEQ(%s)(%s)=%d\n", buf, value, echo[ifl]);
+	}
+	return 1;
+}
+
+static TAG_CALLBACK(spp_r2_hex) {
+	int i;
+	for(i = 0; buf[i]; i++) {
+		if (buf[i] >= '0' && buf[i] <= '9') {
+			int b;
+			unsigned int ch;
+			b = buf[i + 2];
+			buf[i + 2] = '\0';
+			sscanf(buf + i, "%02x", &ch);
+			r_cons_printf ("%c", ch);
+			buf[i + 2] = b;
+			buf = buf + 2;
+		}
+	}
+	return 0;
+}
+
+static TAG_CALLBACK(spp_r2_grepline) {
+	FILE *fd;
+	char b[1024];
+	char *ptr;
+	int line;
+
+	if (!state->echo[state->ifl]) return 1;
+	ptr = strchr(buf, ' ');
+	if (ptr) {
+		*ptr= '\0';
+		fd = fopen (buf, "r");
+		line = atoi (ptr+1);
+		if (fd) {
+			while (!feof (fd) && line--) {
+				if (!fgets (b, 1023, fd)) {
+					break;
+				}
+			}
+			fclose (fd);
+			r_cons_printf ("%s", b);
+		} else {
+			fprintf(stderr, "Unable to open '%s'\n", buf);
+		}
+	}
+	return 0;
+}
+
+static TAG_CALLBACK(spp_r2_else) {
+	state->echo[state->ifl] = state->echo[state->ifl] ? 0 : 1;
+	return 0;
+}
+
+static TAG_CALLBACK(spp_r2_ifnot) {
+	spp_r2_if (state, out, buf);
+	spp_r2_else (state, out, buf);
+	return 1;
+}
+
+static TAG_CALLBACK(spp_r2_ifin) {
+	char *var, *ptr;
+	if (!state->echo[state->ifl]) {
+		return 1;
+	}
+	ptr = strchr (buf, ' ');
+	state->echo[state->ifl + 1] = 0;
+	if (ptr) {
+		*ptr='\0';
+		var = getenv(buf);
+		if (strstr (ptr + 1, var)) {
+			state->echo[state->ifl + 1] = 1;
+		}
+	}
+	return 1;
+}
+
+static TAG_CALLBACK(spp_r2_endif) {
+	return -1;
+}
+
+static TAG_CALLBACK(spp_r2_default) {
+	if (!state->echo[state->ifl]) {
+		return 0;
+	}
+	if (buf[-1] != ';') { /* commented tag */
+		fprintf (stderr, "WARNING: invalid command: '%s' at line %d\n", buf, state->lineno);
+	}
+	return 0;
+}
+
+#if HAVE_SYSTEM
+static FILE *spp_r2_pipe_fd = NULL;
+#endif
+
+static TAG_CALLBACK(spp_r2_pipe) {
+#if HAVE_SYSTEM
+	spp_r2_pipe_fd = popen (buf, "w");
+#endif
+	return 0;
+}
+
+static char *spp_r2_switch_str = NULL;
+
+static TAG_CALLBACK(spp_r2_switch) {
+	char *var = spp_r2_var_get (buf);
+	if (var) {
+		spp_r2_switch_str = strdup (var);
+	} else {
+		spp_r2_switch_str = strdup ("");
```

Comment:
maybe place a `strdup` check?

Reply:
i dont think its needed in case of being null something really terrible should be going on and the worst case we will get the var assigned to null